src/authentication/authentication.service.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  GoneException,
+  Injectable,
+  Logger,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { UsersService } from '../users/users.service';
+import { JwtService } from '@nestjs/jwt';
+import { RefreshTokenIdsStorage } from './refresh-token-ids-storage';
+import { JwtRefreshTokenStrategy } from './strategy/jwt-refresh-token.strategy';
+import { IResponse } from '../shared/response.interface';
+import { LoginDto } from '../users/dto/create-user.dto';
+import {
+  IUser,
+  IUserToken,
+  verificationTypes,
+} from './authentication.interface';
+import { Authentication } from './entities/authentication.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { generateOTPCode } from '../utility/tokenGenerator';
+import { VerifyLoginDto } from './authentication.dto';
+import { NotificationService } from '../shared/notification/notification.service';
+
+@Injectable()
+export class AuthenticationService {
+  private readonly logger = new Logger(JwtRefreshTokenStrategy.name);
+  constructor(
+    private usersService: UsersService,
+    private jwtService: JwtService,
+    private refreshTokenIdsStorage: RefreshTokenIdsStorage,
+    @InjectRepository(Authentication)
+    private readonly authenticationRepository: Repository<Authentication>,
+    private readonly notificationService: NotificationService,
+  ) {}
+
+  async signIn(loginDto: LoginDto): Promise<IResponse<IUser>> {
+    const { email, password } = loginDto;
+
+    const user =
+      await this.usersService.findUserIncludingPasswordByEmail(email);
+
+    if (!user) {
+      throw new UnauthorizedException('Invalid username or password');
```

Comment:
If we are only doing email and no username then it should be email.

Reply:
You are right